src/thor/route_action.cc

```diff
@@ -66,7 +67,12 @@ namespace valhalla {
     // If path is not found try again with relaxed limits (if allowed)
     if (path.empty()) {
       if (cost->AllowMultiPass()) {
-        // 2nd pass. Less aggressive hierarchy transitioning.
+        // 2nd pass. Less aggressive hierarchy transitioning, and retry with more candidate edges(filterd by heading in loki).
+
+        // add filtered edges to candidate edges for origin and destination
+        origin.edges.insert(std::end(origin.edges), std::begin(origin.filtered_edges), std::end(origin.filtered_edges));
```

Comment:
you dont need the `iterator` header and you can just do `origin.edges.end()` or `origin.filtered_edges.begin()` directly